### PR TITLE
Backend/feature/coordinates to timezone

### LINKED ISSUE #795: Function to convert (lat, lng) into timezone
Need a function to convert (longitude, latitude) into a timezone, so that when users input their address (which is used to derive (longitude, latitude)), we can infer their timezone. This timezone would then be used to view all time data through, such as the time of events.

### PR DESCRIPTION
**Pull request by OmerBaddour**
Added function to utils, and test_utils.py

### CODE REVIEW

File: app/backend/requirements.txt [hunk 1]
```diff
@@ -17,3 +17,4 @@ pyyaml
 requests
 Shapely
 sqlalchemy
+timezonefinder[numba]
```
Comment (aapeliv) [inline app/backend/requirements.txt:20]: This file should be sorted

File: app/backend/src/couchers/utils.py [hunk 2]
```diff
@@ -96,6 +97,16 @@ def get_coordinates(geom):
     return (shp.y, shp.x)
 
 
+def to_timezone_lng_lat(lng, lat):
+    """
+    Returns the timezone name corresponding the (lng, lat) input
+    The timezone names are defined in this timezone database:
+    https://www.iana.org/time-zones
+    """
+    tf = TimezoneFinder()
```
Comment (aapeliv) [inline app/backend/src/couchers/utils.py]: Does this reload all of TF every time this function is called? Do we need to create a cached `TimezoneFinder` object and call that each time we want to do the actual computation?

Does this function throw an exception on failure? When can it fail?

File: app/backend/src/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+@pytest.fixture(autouse=True)
+def _(testconfig):
+    pass
+
+
+def test_to_timezone_lng_lat():
+    tf = TimezoneFinder()
+    lat, lng = 52.5061, 13.358
+    assert tf.timezone_at(lng=lng, lat=lat) == "Europe/Berlin"
```
Comment (aapeliv) [inline app/backend/src/tests/test_utils.py]: You need to test the actual function in `utils.py`, not the TimezoneFinder module (I hope they have their own tests!)

**Comment by aapeliv** [comment]
@OmerBaddour: how're you doing on this?

**Comment by aapeliv** [comment]
Also, please make it `(lat, lng)`, we live in EPSG4326

**Comment by OmerBaddour** [comment]
Had a busy week will work on this today

File: app/backend/src/couchers/utils.py (hunk 2 above)
Comment (OmerBaddour) [reply app/backend/src/couchers/utils.py]: I can initialize with `TimezoneFinder(in_memory=True)` to [cache the relevant data](https://timezonefinder.readthedocs.io/en/latest/1_usage.html#initialisation)

File: app/backend/src/couchers/utils.py [hunk 4]
```diff
@@ -96,6 +97,19 @@ def get_coordinates(geom):
     return (shp.y, shp.x)
 
 
+def to_timezone(lat, lng):
+    """
+    Returns the timezone name corresponding the lat, lng input, or None if the input is
```
Comment (aapeliv) [inline app/backend/src/couchers/utils.py]: if the input is?

File: app/backend/src/couchers/utils.py [hunk 5]
```diff
@@ -96,6 +97,19 @@ def get_coordinates(geom):
...
+    The timezone names are defined in this timezone database:
+    https://www.iana.org/time-zones
+    """
+    tf = TimezoneFinder(in_memory=True)
+    try:
+        return tf.timezone_at(lat=lat, lng=lng)
+    except ValueError as ve:
+        print("Error: to_timezone():", ve)
```
Comment (aapeliv) [inline app/backend/src/couchers/utils.py]: This seems incomplete.

Does TF not have its own exception we could catch?

File: app/backend/src/tests/test_utils.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+
+def test_to_timezone():
+    # test valid lat, lng
+    lat, lng = 52.5061, 13.358
+    assert to_timezone(lat=lat, lng=lng) == "Europe/Berlin"
+
+    # test invalid lat, lng
+    lat, lng = -91, -181
+    assert to_timezone(lat=lat, lng=lng) == None
```
Comment (aapeliv) [inline app/backend/src/tests/test_utils.py]: Use `is` when comparing to `None`

File: app/backend/src/couchers/utils.py [hunk 7]
```diff
@@ -96,6 +97,19 @@ def get_coordinates(geom):
     return (shp.y, shp.x)
 
 
+def to_timezone(lat, lng):
```
Comment (aapeliv) [inline app/backend/src/couchers/utils.py]: Let's call this function `timezone_at_coordinate`

**Review by aapeliv — CHANGES_REQUESTED** [review]

File: app/backend/src/couchers/utils.py (hunk 5 above)
Comment (OmerBaddour) [reply app/backend/src/couchers/utils.py]: I followed the sequence of methods and functions that are called, and only a `ValueError` can be raised.

File: app/backend/src/couchers/utils.py [hunk 8]
```diff
@@ -96,6 +97,19 @@ def get_coordinates(geom):
...
+    """
+    Returns the timezone name corresponding the lat, lng input, or None if the input does not correspond to a timezone or is invalid.
+    The timezone names are defined in this timezone database:
+    https://www.iana.org/time-zones
+    """
+    tf = TimezoneFinder(in_memory=True)
+    try:
+        return tf.timezone_at(lat=lat, lng=lng)
+    except ValueError as ve:
+        print("Error: timezone_at_coordinate():", ve)
```
Comment (aapeliv) [inline app/backend/src/couchers/utils.py:110]: Please don't `print`, you can `logger.debug` if you wish.

Also please explicitly `return None`, it's just cleaner that way!

**Review by aapeliv — COMMENTED** [review]
Looks good, small nit.

Is there any place within the WGS84 bounds that also returns `None`, or is that only outside the bounds?

**Pull request closed without merging**